### PR TITLE
reset dynamo cache before each test

### PR DESCRIPTION
**Pull request by shunting314**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #126586

In https://github.com/pytorch/pytorch/issues/125967, we found test results depend on test order. The root cause is due to earlier tests populate dynamo cache and affect the later tests.

This PR clear dynamo cache before each unit test so we get more deterministic result for unit test

### CODE REVIEW

**Review by jansel — APPROVED** [review]

File: torch/testing/_internal/common_utils.py
Comment (huydhn) [inline torch/testing/_internal/common_utils.py:2909]: I vaguely remember that this causes a perf hit on CI, so it couldn't be rolled out before.  @zou3519 would know more

Comment (zou3519) [reply torch/testing/_internal/common_utils.py:2909]: We do need to do this (reset dynamo cache). I'm not sure about the performance impact on inductor tests, but it is hopefully not bad because we do not run that many inductor tests.

@shunting314 can you get the tests to pass in this PR so that we can see the before/after test timing numbers?

Comment (shunting314) [reply torch/testing/_internal/common_utils.py:2909]: yea. 2 of the failures are due to previously fail tests now pass. I removed the file marking the test failure.

For the other failures, I'm not able to repro. E.g., for https://github.com/pytorch/pytorch/actions/runs/9135411752/job/25122927855 , I run 
```
PYTORCH_TEST_WITH_INDUCTOR=1 python test/functorch/test_ops.py -k test_extremal_numerics_l1_loss_cpu
```
And it passes on my devserver..

Any suggestions for this?

Comment (zou3519) [reply torch/testing/_internal/common_utils.py:2909]: it might have to do with python version. Try to skip these for now. Also, the `keep-going` label will make all tests run -- it's possible you have more failures than just these two

Comment (shunting314) [reply torch/testing/_internal/common_utils.py:2909]: Since the PR is clean now (the previous failed tests shows up as flaky and non-blocking in CI), I'll merge it.

Comment (zou3519) [reply torch/testing/_internal/common_utils.py:2909]: > Since the PR is clean now (the previous failed tests shows up as flaky and non-blocking in CI), I'll merge it.

Yup go for it

**Comment by shunting314** [comment]
hmm, the test failures blocks the merge even though CI is green before I try to merge.

I'll skip those tests for now.

**Pull request closed without merging**
